File: 1022-sum-of-root-to-leaf-binary-numbers/1022-sum-of-root-to-leaf-binary-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumRootToLeaf(TreeNode* root, int sum = 0) {
        if(root == nullptr) return 0;
        sum = sum * 2 + root->val;
        return root->left == root->right ? 
            sum : sumRootToLeaf(root->left, sum) + sumRootToLeaf(root->right, sum);
    }
};

File: 1022-sum-of-root-to-leaf-binary-numbers/1022-sum-of-root-to-leaf-binary-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumRootToLeaf(self, root: Optional[TreeNode], ans: int = 0) -> int:
            if not root: return 0;
            ans = ans * 2 + root.val;
            return ans if root.left == root.right else self.sumRootToLeaf(root.left, ans) + self.sumRootToLeaf(root.right, ans);